capstone-app/src/app/modules/history/wiki-results.service.ts

```diff
@@ -82,50 +86,70 @@ export class WikiResultsService {
       parsedText = firstPartOfString.substring(startIndex, endIndex);
     }
-    else {
-      console.error('The city page was found, but unfortunately there was no history paragraph found!');
-    }
     const paragraphs = parsedText.split('<p>');
     if (paragraphs.length > 9) {
       if (paragraphs[0].startsWith('</h2')) {
-        parsedText = paragraphs.slice(1,10).join('');
+        parsedText = paragraphs.slice(1,10).join('----');
       }
       else {
-        parsedText = paragraphs.slice(0,9).join('');
+        parsedText = paragraphs.slice(0,9).join('----');
       }
     }
+    else {
+      parsedText = paragraphs.join('----');
+    }
+    const regexForSourceNeeded = /<sup class="noprint Inline-Template Template-Fact"[\s\S]*?<\/sup>/gi;
+    parsedText = parsedText.replace(regexForSourceNeeded, '');
     const furtherReading = this.findHrefs(parsedText, language);
-    let history = parsedText.split(/<h3>.*?<\/h3>/g).join('');
-    const regexForULists = /<ul[\s\S]+<\/ul>/gi;
-    const regexForOLists = /<ol[\s\S]+<\/ol>/gi;
-    const regexForTables = /<table[\s\S]+<\/table>/gi;
-    const regexForCPUStats = /<!--[\s\S]*-->/gi;
-    const regexForAllTags = /<.*?>/gi;
+    let tempHistory = parsedText;
+    const regexForCatLinks = /class="catlinks[\s\S]*?<\/div>/gi;
+    const regexForDabLinks = /<div class="dablink[\s\S]*?<\/div>/gi;
+    const regexCaptions = /<a.*?<\/div><\/div><\/div>/gi;
+    const regexParsing = /\.mw.*?}}/gi;
+    const regexForH3 = /<h3>.*?<\/h3>/gi;
+    const regexForULists = /<ul[\s\S]+?<\/ul>/gi;
+    const regexForOLists = /<ol[\s\S]+?<\/ol>/gi;
+    const regexForTables = /<table[\s\S]+?<\/table>/gi;
+    const regexForNotes = /<div role="note"[\s\S]+?<\/div>/gi;
+    const regexForCPUStats = /<!--[\s\S]*?-->/gi;
+    const regexForCitationNeeded = /<span[\s\S]*?<\/span>/gi;
+    const regexForRows = /<tr[\s\S]*?<\/tr>/gi;
+    const regexForColumns = /<td[\s\S]*?<\/td>/gi;
+    const regexForAllTags = /<[\s\S]*?>/gi;
     const regexForContentListDiv = /<div id="toc"[\s\S]*?<\/div>/gi;
     const regexForExtraColorAttributes = /\d*&.*?;/gi;
     const regexForRefrenceNumbers = /\[[\s\S]+?]/gi;
     const regexForAnyRemainingCSSStyling = /.mw-[\s\S]+?}/gi;
-    history = history.replace(regexForOLists, '').replace(regexForULists, '')
-    .replace(regexForTables, '').replace(regexForCPUStats, '').replace(regexForContentListDiv, '')
-    .replace(regexForRefrenceNumbers, '').replace(regexForAllTags, '').replace(regexForExtraColorAttributes, '')
-    .replace(regexForAnyRemainingCSSStyling, '');
+    const regexForDivStyleTags = /<div style=[\s\S]*?<\/div>/gi;
+    tempHistory = tempHistory.replace(regexCaptions, '').replace(regexParsing, '').replace(regexForCatLinks, '>');
+    tempHistory = tempHistory.replace(regexForOLists, '').replace(regexForULists, '').replace(regexForDabLinks, '');
+    tempHistory = tempHistory.replace(regexForTables, '').replace(regexForNotes, '');
```

Comment:
Is there a reason to switch to multiple reassignments?

Reply:
Yes, it makes debugging much more manageable. I can put print statements in between all of them. Otherwise I can't tell where the over/ under parsing is occurring.